eng/pipelines/templates/stages/archetype-sdk-client.yml

```diff
@@ -38,9 +38,26 @@ parameters:
 - name: TestEnv
   type: object
   default: []
+- name: CMakeTestOptions
+  type: object
+  default: []
+- name: CMakeTestDependencies
+  type: string
+  default: ''
 
 stages:
+  - ${{ if eq(variables['System.TeamProject'], 'public') }}:
```

Comment:
Let's leave this as a separate stage for now since the other test matrix is crowded enough already :)